qiskit/circuit/parameter.py

```diff
@@ -155,7 +154,7 @@ def _hash_key(self):
         # expression, so its full hash key is split into `(parameter_keys, symbolic_expression)`.
         # This method lets containing expressions get only the bits they need for equality checks in
         # the first value, without wasting time re-hashing individual Sympy/Symengine symbols.
-        return (self._name, self._uuid)
```

Comment:
This change might affect code using picking. Would it be possible to keep the tuple shape the same?